src/nxdoc/nuxeo-server/platform-services/comments.md

```diff
@@ -20,7 +20,73 @@ Comments (and so annotations) are linked to a specific document (their parent).
 
 ## Core Implementation
 
-### Storage
+### Current implementation - TreeCommentManager
+
+#### Storage
+
+Comments in Nuxeo are stored as regular documents of type `Comment`.</br>
+Comments hold:
+
+- parent id (legacy purpose)
+- ancestor ids (legacy purpose)
+- author
+- text
+- creation date
+- modification date
+
+[`Comment`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Comment.html) interface has been created to handle comments in Nuxeo. The interface is used as based type in [comment service](#java-api) and also defines how comments are serialized in JSON in order to use them with the [REST API](#json-entity-format).
+
+Annotations are stored as regular document of type `Annotation` which inherit from `Comment`. They hold in addition of comments:
+
+- blob xpath
+
+Like comments, annotations have the [`Annotation`](http://community.nuxeo.com/api/nuxeo/latest/javadoc/org/nuxeo/ecm/platform/comment/api/Annotation.html) interface to use them.
+
+Comments and annotations can have the facet `ExternalEntity`. When they have this facet, it can hold in addition:
+
+- external entity id
+- external entity
+- external entity origin</br>
+With this facet, it is possible to handle external content such as annotations made in [ARender]({{page space='nxdoc' page='nuxeo-arender-connector'}}) for instance.
+
```

Comment:
I am wondering if we refactor the part of the common things (between the tree and property), I mean we have the same definition on type: `Comment` and `Annotation` we provide the same properties (main difference is for the Tree, parentId and ancestor are kept for backward compatibility). to accomplish it one of idea is to give definition and common part at beginning and add an implementation section